tests/test.py

```diff
@@ -5145,6 +5145,383 @@ class TestPostgresSQLPreparedStatements(TestSQLPreparedStatements, BaseTokenizat
+        }, literal_binds=False)
+
+        args = []
+        arg = 'test_prepared_sql_statements__searchable'
+        args.append(arg)
+        self.set_arg(arg_name=arg, engine=self.engine2, value=search_term)
+
+        self.execute_prepared(prepared_name='delete_data_by_field', engine=self.engine2, args=args)
+
+        rows = self.execute_prepared_fetch(prepared_name='select_data_by_field', engine=self.engine2, args=args)
+        self.assertEqual(len(rows), 0)
+
+    def testSearchableEncryptionWithDeallocate(self):
+        base.metadata.create_all(self.engine_raw, [self.test_prepared_sql_statements_table])
+        self.engine1.execute(self.test_prepared_sql_statements_table.delete())
+
+        context = self.get_test_prepared_sql_statements_table_context()
+        search_term = context['searchable']
+
+        # Insert searchable data and some additional different rows
+        _, columns_order = self.prepare(prepared_name='insert_data', engine=self.engine2,
+                                       query=self.test_prepared_sql_statements_table.insert(),
+                                       data_types=self.prepared_sql_statements_table_data_types)
+
+        args = []
+        for key in columns_order:
+            arg = 'test_prepared_sql_statements__{}'.format(key)
+            args.append(arg)
+            self.set_arg(arg_name=arg, engine=self.engine2, value=context[key])
+
+        self.execute_prepared(prepared_name='insert_data', engine=self.engine2, args=args)
+
+        extra_rows_count = 5
+        while extra_rows_count != 0:
+            new_data = base.get_pregenerated_random_data().encode('utf-8')
+            if new_data != search_term:
+                new_id = context['id'] + extra_rows_count
+                self.set_arg(arg_name='test_prepared_sql_statements__searchable', engine=self.engine2, value=new_data)
+                self.set_arg(arg_name='test_prepared_sql_statements__id', engine=self.engine2, value=new_id)
+                self.execute_prepared(prepared_name='insert_data', engine=self.engine2, args=args)
+                extra_rows_count -= 1
+
+        query = sa.select(self.test_prepared_sql_statements_table).where(
+            self.test_prepared_sql_statements_table.c.searchable == search_term)
+
+        self.prepare(prepared_name='select_data_by_field', engine=self.engine2, query=query, data_types={
+            'searchable': 'bytea'
+        }, literal_binds=False)
+
+        args = []
+        arg = 'test_prepared_sql_statements__searchable'
+        args.append(arg)
+        self.set_arg(arg_name=arg, engine=self.engine2, value=search_term)
+
+        rows = self.execute_prepared_fetch(prepared_name='select_data_by_field', engine=self.engine2, args=args)
+        self.assertEqual(len(rows), 1)
+
+        # deallocate prepared statement from DB and delete statement from session registry
+        self.deallocate(prepared_name='select_data_by_field', engine=self.engine2)
```

Comment:
reading this test pushed to think about the case when driver:
1. set variables
2. prepared query with placeholders
3. executed
4. deallocated
5. prepared same query
6. executed with previously set values

IMHO we should test this case too to be sure that leaving previously set values handled properly in the next prepared statement
What do you think?